### PR TITLE
Input variables in tasks

### LINKED ISSUE #4758: Task Runner: Provide input parameters
- VSCode Version: 0.10.11
- OS Version: Mac OS X El Capitan

I would like to provide input parameters when a task is executed. How can I accomplish this?

Cheers,
Trevor Sullivan
Microsoft MVP: PowerShell

### PR DESCRIPTION
**Pull request by alexr00**
Fixes #4758

### CODE REVIEW

**Comment by alexr00** [comment]
@isidorn, could you take a look at the configurationResolver* files? I made some significant changes to them and I want to make sure you're ok with them.

File: src/vs/workbench/services/configurationResolver/common/configurationResolver.ts [hunk 1]
```diff
@@ -24,9 +24,22 @@ export interface IConfigurationResolverService {
...
-	 * Recursively resolves all variables (including commands) in the given config and returns a copy of it with substituted values.
+	 * Recursively resolves all variables (including commands and user input) in the given config and returns a copy of it with substituted values.
 	 * If a "variables" dictionary (with names -> command ids) is given,
 	 * command variables are first mapped through it before being resolved.
 	 */
-	resolveWithCommands(folder: IWorkspaceFolder, config: any, variables?: IStringDictionary<string>): TPromise<any>;
+	resolveWithInteraction(folder: IWorkspaceFolder, config: any, variables?: IStringDictionary<string>): TPromise<any>;
 }
+
+export enum ConfiguredInputType {
```
Comment (isidorn) [inline src/vs/workbench/services/configurationResolver/common/configurationResolver.ts]: Use `const enum`

File: src/vs/workbench/services/configurationResolver/common/configurationResolver.ts [hunk 2]
```diff
@@ -24,9 +24,22 @@ export interface IConfigurationResolverService {
...
+	prompt,
```
Comment (isidorn) [inline src/vs/workbench/services/configurationResolver/common/configurationResolver.ts]: We usualy capitalise the first letter of enum values

**Comment by isidorn** [comment]
@alexr00 I took a look at the changes you made to the `configuratoinResolver` files and they look good to me overall. Great job! I also see you added some tests, so I would just verify:
- [ ] all the tests are green -> this seems to be good
- [ ] variables still get resolved for all the cases
  - [ ] `${workspaceRoot}`
  - [ ] `${end:PATH}`
  - [ ] `${command:SOME_COMMAND}`
  - [ ] `${config:editor.lineHeight}`

Apart from that I would add a bit more comments once you find the time. I know it was like that before, but we could improve this.
Added some minor comments inline

@weinand might also want to take a look at the changes.

**Comment by weinand** [comment]
@alexr00 Please make sure that variables are resolved correctly in the WSL and dev container setups.

**Pull request merged**